### PR TITLE
Added an ability for admin to add new articles for lifestyle and dealbook

### PR DESCRIPTION
**Pull request by SwaTiK4931**

### CODE REVIEW

File: sports_hub/sports_hub/Views/AdminPage/AddNewLifestyleArticle.cshtml [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        function hideStartPicturePopOuts() {
+            document.getElementById("containerPicturePopOutsStart").classList.toggle("hidden", true);
+        }
+
+        function submitForm() {
+            document.getElementById("Form").submit();
+        }
+    </script>
+    <script>
+    // scripts for drag and drop image
```
Comment (XashDM) [inline sports_hub/sports_hub/Views/AdminPage/AddNewLifestyleArticle.cshtml:187]: format

File: sports_hub/sports_hub/Controllers/AdminPageController.cs [hunk 2]
```diff
@@ -50,6 +50,79 @@ public IActionResult Lifestyle()
...
+                Article.Content = "";
+                ModelState.Remove("Article.Content");
+            }
+
+            if (ModelState.IsValid)
+            {
+                Article.Image = imageData;
+
+                _context.ArticlesDealbook.Add(Article);
+                _context.SaveChanges();
```
Comment (XashDM) [inline sports_hub/sports_hub/Controllers/AdminPageController.cs:116]: Move this to AricleService - it should handle deal books and so on

File: sports_hub/sports_hub/Controllers/AdminPageController.cs [hunk 3]
```diff
@@ -50,6 +50,79 @@ public IActionResult Lifestyle()
...
+                Article.Content = "";
+                ModelState.Remove("Article.Content");
+            }
+
+            if (ModelState.IsValid)
+            {
+                Article.Image = imageData;
+
+                _context.ArticlesLifestyle.Add(Article);
+                _context.SaveChanges();
```
Comment (XashDM) [inline sports_hub/sports_hub/Controllers/AdminPageController.cs:77]: Move to article service

**Review by XashDM — COMMENTED** [review]
Lets discuss those changes

File: sports_hub/sports_hub/Views/AdminPage/AddNewLifestyleArticle.cshtml (hunk 1 above)
Comment (SwaTiK4931) [reply sports_hub/sports_hub/Views/AdminPage/AddNewLifestyleArticle.cshtml:187]: remove comment?

File: sports_hub/sports_hub/Services/Interfaces/IArticlesService.cs [hunk 4]
```diff
@@ -21,5 +22,7 @@ public interface IArticlesService
         public void ChangeArticleTeam(int article_id, int id_team);
 
         public AddNewArticleViewModel CreateNewArticleViewModel(string Category);
+
+        public byte[] ConvertIFormFileToBytes(IFormFile File);
```
[the remark this answers is not in the document]
Comment (SwaTiK4931) [reply sports_hub/sports_hub/Services/Interfaces/IArticlesService.cs:26]: I need to move it to controller as private method?

**Comment by SwaTiK4931** [comment]
I've implemented TPH approach - Lifestyle and Dealbook articles are now stored in one table, and moved adding articles to article service

**Review by XashDM — APPROVED** [review]
Great work, you can merge - only thing that is to add article to same hierarchy, create item for that in backlog

**Pull request merged**